pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
The adoption of a RS out of nowhere will affect the proportional scaling code but not `NewRSNewReplicas` because we are not scaling up if `currentPodCount >= maxTotalPods`.